neuro_san/session/async_direct_agent_session.py

```diff
@@ -71,6 +74,18 @@ def __init__(self,
         self.request_id: str = None
         if metadata is not None:
             self.request_id = metadata.get("request_id")
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def _log_agent_network_usage(self, operation: str):
+        """
+        Logs the usage of the agent network if enabled.
+        :param operation: The operation being performed (e.g., "function", "connectivity", "streaming_chat").
+        """
+        if self.LOG_AGENT_NETWORK_USAGE:
+            self.logger.info("Agent network '%s' (ID: %d) used for operation: %s",
+                             self.agent_network.get_network_name(),
+                             id(self.agent_network),
+                             operation)
```

Comment:
More debug logging to track which AgentNetwork instances are really used for API methods.